### PR TITLE
Use promise based dmn moddle

### PR DESCRIPTION
**Pull request by MaxTru**
Related to #514 

* bump dmn-moddle
* Use promise-based dmn-moddle APIs
* bump test suite dev-dependencies

DEPRECATION: the <import.parse.complete> event is now invoked with
    
    {
     error,
     definitions,
     elementsById,
     references,
     warnings
    }
    
The old `context` is deprecated and will be removed with a future release. Assessing the old `context` property will log an error but is still possible:
    
    {
      error,
      definitions,
      context: { elementsById, references, warnings }
    }

After this PR, I will promisify the public dmn-js APIs

### CODE REVIEW

**Comment by barmac** [comment]
Is it in fact a breaking change? The old flow still works but it generates a warning, correct?

**Comment by MaxTru** [comment]
> Is it in fact a breaking change? The old flow still works but it generates a warning, correct?

True, so it is just a deprecation => I will adjust the commit message and PR description. 

Thanks!

**Review by barmac — APPROVED** [review]
LGTM

**Review by nikku — COMMENTED** [review]
Forgot to press the "Submit review" button yesterday.

Thanks for following up @barmac.

File: packages/dmn-js-shared/src/base/Manager.js [hunk 1]
```diff
@@ -275,30 +275,19 @@ export default class Manager {
...
-          error: error,
-          xml: xml
-        });
-      } catch (e) {
-        console.error('error in saveXML life-cycle listener', e);
-      }
+      self._emit('saveXML.done', result);
 
-      done(error, xml);
+      done(result.error, result.xml);
```
[the remark this answers is not in the document]
Comment (MaxTru) [reply packages/dmn-js-shared/src/base/Manager.js:290]: Yep :+1:

**Pull request merged**